src/test/java/io/lettuce/core/cluster/pubsub/RedisClusterPubSubConnectionIntegrationTests.java

```diff
@@ -126,13 +133,55 @@ void subscribeToShardChannelViaReplica() {
         int clusterKeyslot = connection.sync().clusterKeyslot(shardChannel).intValue();
         String thisNode = connection.getPartitions().getPartitionBySlot(clusterKeyslot).getNodeId();
 
-        RedisPubSubAsyncCommands<String, String> replica =
-                pubSubConnection.async().nodes(node -> thisNode.equals(node.getSlaveOf())).commands(0);
+        RedisPubSubAsyncCommands<String, String> replica = pubSubConnection.async()
+                .nodes(node -> thisNode.equals(node.getSlaveOf())).commands(0);
         replica.ssubscribe(shardChannel);
 
         Wait.untilEquals(shardChannel, connectionListener.getShardChannels()::poll).waitOrTimeout();
     }
 
+    @Test
+    @EnabledOnCommand("SSUBSCRIBE")
+    void publishToShardChannel() throws Exception {
+        pubSubConnection.addListener(connectionListener);
+        pubSubConnection.async().ssubscribe(shardChannel);
+        Wait.untilEquals(shardChannel, connectionListener.getShardChannels()::poll).waitOrTimeout();
+
+        pubSubConnection.async().spublish(shardChannel, shardMessage);
+        Wait.untilEquals(shardChannel, connectionListener.getShardChannels()::poll).waitOrTimeout();
+        Wait.untilEquals(shardMessage, connectionListener.getMessages()::poll).waitOrTimeout();
+    }
+
+    @Test
+    @EnabledOnCommand("SSUBSCRIBE")
+    void publishToShardChannelViaDifferentEndpoints() throws Exception {
```

Comment:
What are we really testing here :)?

Reply:
two spublish ends up with different endpoints successfully and we are receivng msgs back with only/same connection listener.

Reply:
So when you say "endpoints" you mean "shard channels"? Or do you mean that the second shard channel is part of another shard and we would (behind the scenes) send the message to another node?

Can we add this as a comment to the test? I'd like us to be clear about what the goal of the test is.

Reply:
endpoints means the service that you reach out ,, basically ip:port or host:port.
in this test case, Redis cluster with two upstream nodes bring their individual endpoints.
i will keep in mind to add some comments with future commits/changes.

Reply:
Yes, this is also my understanding of what endpoints means, but then in the code I see we use the same connection. So my assumption now is that we rely on this connection object switching to another node (behind the scenes) but this is not mentioned implicitly. And thus I was wondering if I understand you correctly